### ISSUE TITLE
Request socket can only be acquired once in two phases

### ISSUE THREAD

**Issue by fffonion** [openresty/stream-lua-nginx-module#101]
Hi,
When I try to acquire the raw request socket for two times in different phases, lua gives me a "duplicate call" error in the second phase. 

Tested both on version `0.0.3` and `0.0.4rc3`

Minimal config:

```lua
stream {
    server {
        listen 20000;
        preread_by_lua_block {
            assert(ngx.req.socket())
        }

        content_by_lua_block {
            assert(ngx.req.socket())
        }
    }
}
```

I see there's a counter for every request that counts the total times raw socket is acquired. Is this an expected behavior for the stream sub system? It makes sense in the http system because we definitely want to write to request socket only once in the content phase not access or rewrite phase.

The reason I found this is I was trying to implement the `ngx.req.socket():peak()` function. If we acquire the socket in **preread** phase, we will then be not able to acquire it again in the **content** phase.

Sorry for sending the incomplete issue before I finish at first.

**Comment by dndx** [comment]
Hello,

Yes this is by design. It is not safe to acquire the socket more than once due to the `__gc` metamethod it contains to cleanup properly should user code loses reference. You can save the object inside `ngx.ctx` and retrieve it in later phase.

**Comment by fffonion** [comment]
@dndx Thank you!
